docs/spec.md

```diff
@@ -2633,6 +2633,50 @@ planning to address this in
 
 &nbsp;[More Examples](https://github.com/openxla/stablehlo/tree/main/stablehlo/tests/interpret/dot_general.mlir)
 
+### dynamic_iota
+
+#### Semantics
+
+This operation is functionally identical to
+[iota](https://github.com/openxla/stablehlo/blob/main/docs/spec.md#iota)
+op, but the result shape is specified dynamically via `output_shape`.
+
+#### Inputs
+
+| Label | Name             | Type                                                                               | Constraints |
+|-------|------------------|------------------------------------------------------------------------------------|-------------|
+| (I1)  | `output_shape`   | 1-dimensional tensor constant of type `si64`                                       | (C1), (C2)  |
+| (I2)  | `iota_dimension` | `si64`                                                                             | (C1)        |
+
+#### Outputs
+
+| Name     | Type                                                                              | Constraints |
+|----------|-----------------------------------------------------------------------------------|-------------|
+| `result` | tensor of integer, floating-point, or complex type or per-tensor quantized tensor |     (C2)    |
+
+#### Constraints
+
+* (C1) `0 <= iota_dimension < size(output_shape)`.
+* (C2) `rank(result) = size(output_shape)`.
+
+#### Examples
+
+```mlir
+
+%output_shape = stablehlo.constant dense<[4, 5]> : tensor<2xi64>
+%result = "stablehlo.dynamic_iota"(%output_shape) {
+  iota_dimension = 0 : i64
+} : (tensor<2xi64>) -> tensor<4x5xi64>
```

Comment:
do we need an example with dynamic shapes (result  tensor<?x?xi64>) ? 
reference interpreter doesn't support dynamic shapes for now,  I used static shapes in example.

cc @sdasgup3 @ghpvnist

Reply:
IMO that would be preferable, since that's how the op is meant to be used. IMO the example does not need to be something the interpreter supports, and when the interpreter is updated to handle dynamic shapes we are unlikely to revisit this example.

Reply:
Let's keep the static example for now to keep docs in sync with the reference interpreter. Once dynamic shapes are supported, we can update the spec with dynamic example and then add a test for it. We can consider keeping the static version as an extra example in the test.

Reply:
My 2cents:

With different dynamic ops lined up for specification, I think it would be better to spec the op with intended examples. IMO it is OK for the the spec test not to follow interpreter guidelines (1 and 4 below) before dynamic op interpretation is supported. 

For example, From https://github.com/openxla/stablehlo/blob/main/docs/reference_checklist.md (7)

```
1 Add a link to stablehlo/tests/interpret/<op_mnemonic>.mlir to the "Examples" section (e.g. [More Examples](https://github.com/openxla/stablehlo/blob/main/docs/spec.md#add)).
2 Make sure the spec only has 1 example.
3 Make sure the spec example follows the [testing guidelines](https://github.com/openxla/stablehlo/blob/main/docs/reference.md#testing-guidelines).
4 Make sure the spec example test is interpretable.
5 Make sure the spec example is the same as what is in the ODS.
``` 

1. No "More examples" link as  as dynamic interpretation is not fully supported yet. 
4. Is it fine that we cannot do that. IMO we should not write an example just so that  (4) is honored.  


With that once the dynamic interpretation is supported we have one artifact to update (the interpreter test cases) as opposed to multiple (spec, ODS).

Reply:
Thanks folks, as per offline discussion, keeping the static shape example.